src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/HPackDecoder.cs

```diff
@@ -92,12 +99,10 @@ private enum State : byte
         private byte[] _stringOctets;
         private byte[] _headerNameOctets;
         private byte[] _headerValueOctets;
-        private (int start, int length)? _headerNameRange;
-        private (int start, int length)? _headerValueRange;
+        private HeaderData _headerData;
 
         private State _state = State.Ready;
         private byte[]? _headerName;
-        private int _headerStaticIndex;
```

Comment:
Will the JIT coalesce the zeroings if you move this field next to the others? And if not, should it, and is there an issue tracking this optimization?

Reply:
> I'm not sure if the JIT would be able to recognize this w/o help, as the fields can be arranged in a way that the JIT thinks is best.

At that point the fields are already arranged, and the JIT should see a couple of zeroings to adjacent memory locations, and coalesce them.